api4/system.go

```diff
@@ -364,3 +366,89 @@ func collectMobileTelemetry(c *Context, w http.ResponseWriter, r *http.Request)
 	ReturnStatusOK(w)
 	return
 }
+
+func sendAlertNotification(c *Context, w http.ResponseWriter, r *http.Request) {
+
+	if !c.App.SessionHasPermissionToSendAlerts(c.App.Session) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_JOBS)
+		mlog.Error(fmt.Sprintf("UserId: %v does not have permission to send alerts", c.App.Session.UserId))
+		return
+	}
+
+	notificationReq := model.MobileAlertRequestFromJson(r.Body)
+	if notificationReq == nil {
+		return
+	}
+
+	if notificationReq.PostId == nil || *notificationReq.PostId == "" {
+		mlog.Error("Invalid Request, Mandatory fields are missing - 'postId' ")
+		return
+	}
+
+	postInfo, err := c.App.GetSinglePost(*notificationReq.PostId)
+	if err != nil {
+		mlog.Error(fmt.Sprintf("No valid Post data found for the postId: %v", *notificationReq.PostId))
+		return
+	}
+
+	pushNotificationsEnabled := false
+	if *c.App.Config().EmailSettings.SendPushNotifications {
+		pushServer := *c.App.Config().EmailSettings.PushNotificationServer
+		if license := c.App.License(); pushServer == model.MHPNS && (license == nil || !*license.Features.MHPNS) {
+			mlog.Warn("Push notifications are disabled. Go to System Console > Notifications > Mobile Push to enable them.")
+			pushNotificationsEnabled = false
```

Comment:
Is it good for us to return here if the notifications are disabled to avoid the further lines to be executed?
And also return when notificationReq.SendPushNotification is false?.

Reply:
ya, we can do that

Reply:
changes implemented now